lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -2,40 +2,21 @@
  * Copyright (c) Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License.
  */
-import { Constants, PersistentCacheKeys, StringUtils, AuthorizationCodeRequest, ICrypto, CacheSchemaType, AccountEntity, IdTokenEntity, CredentialType, AccessTokenEntity, RefreshTokenEntity, AppMetadataEntity, CacheManager, CredentialEntity } from "@azure/msal-common";
-import { CacheOptions } from "../config/Configuration";
-import { BrowserAuthError } from "../error/BrowserAuthError";
+
+import { IWindowStorage } from "./IWindowStorage";
 import { BrowserConfigurationAuthError } from "../error/BrowserConfigurationAuthError";
-import { BrowserConstants, TemporaryCacheKeys } from "../utils/BrowserConstants";
+import { BrowserAuthError } from "../error/BrowserAuthError";
+import { BrowserConstants } from "../utils/BrowserConstants";
 
-// Cookie life calculation (hours * minutes * seconds * ms)
-const COOKIE_LIFE_MULTIPLIER = 24 * 60 * 60 * 1000;
+export class BrowserStorage implements IWindowStorage {
 
-/**
- * This class implements the cache storage interface for MSAL through browser local or session storage.
- * Cookies are only used if storeAuthStateInCookie is true, and are only used for
- * parameters such as state and nonce, generally.
- */
-export class BrowserStorage extends CacheManager {
-
-    // Cache configuration, either set by user or default values.
-    private cacheConfig: CacheOptions;
-    // Window storage object (either local or sessionStorage)
     private windowStorage: Storage;
-    // Client id of application. Used in cache keys to partition cache correctly in the case of multiple instances of MSAL.
-    private clientId: string;
 
-    constructor(clientId: string, cacheConfig: CacheOptions) {
-        super();
+    constructor(cacheLocation: string) {
         // Validate cache location
-        this.validateWindowStorage(cacheConfig.cacheLocation);
-
-        this.cacheConfig = cacheConfig;
-        this.windowStorage = window[this.cacheConfig.cacheLocation];
-        this.clientId = clientId;
+        this.validateWindowStorage(cacheLocation);
```

Comment:
This function should catch errors that are thrown when acccessing local/session storage, and throw appropriate error about them not being supported/accessible in the browser.

Reply:
That error should probably recommend using custom storage (or the built-in in memory option, if we decide to provide that).